firmware/common/portapack_persistent_memory.cpp

```diff
@@ -923,5 +931,122 @@ size_t data_size() {
     return sizeof(data_t);
 }
 
+bool debug_dump() {
+    ui::Painter painter{};
+    std::string debug_dir = "DEBUG";
+    std::filesystem::path filename{};
+    File pmem_dump_file{};
+    // create new dump file name and DEBUG directory
+    make_new_directory(debug_dir);
+    filename = next_filename_matching_pattern(debug_dir + "/DEBUG_DUMP_????.*");
+    if (filename.empty()) {
+        return false;
+    }
+    filename = filename.replace_extension(u".TXT");
+    // dump data fo filename
+    auto result = pmem_dump_file.create(filename);  // First freq if no output file
+    if (!result.is_valid()) {
```

Comment:
nit: I think a nicer looking pattern with opening/creating is like this with early return.
```
auto error = file.create...
if (error)
    return;
```

That way it's clear there's an error and you remove a layer of indent.

Reply:
I guess in your case you'd also display the error here before returning.

Reply:
Done, and cleared the copy pasted comment, bad me ^^